packages/frontend/src/features/semanticViewer/components/SidebarViewFields.tsx

```diff
@@ -94,9 +91,22 @@ const SidebarViewFields = () => {
     }
 
     const handleFieldToggle = (
-        field: Pick<SemanticLayerField, 'name' | 'kind' | 'type'>,
+        field:
+            | Pick<SemanticLayerField, 'name' | 'kind'>
+            | Pick<SemanticLayerTimeDimension, 'name' | 'kind' | 'granularity'>,
     ) => {
-        dispatch(toggleField(field));
+        if ('granularity' in field) {
+            return dispatch(toggleTimeDimension(field));
+        }
+
+        switch (field.kind) {
+            case FieldKind.DIMENSION:
+                return dispatch(toggleDimension(field));
+            case FieldKind.METRIC:
+                return dispatch(toggleMetric(field));
+            default:
+                return assertUnreachable(field.kind, 'Unknown field kind');
+        }
```

Comment:
`granularity` in `timeDimension` is nullable, should we just check if `kind === FieldKind.DIMENSION && type === SemanticLayerFieldType.TIME` so that time dimensions always get sent in that query param?

Reply:
good point.. updating